### PR TITLE
Remove hardcoded retryable data functions

### PR DESCRIPTION
**Pull request by yahgwai**

### CODE REVIEW

File: src/lib/message/L1ToL2MessageGasEstimator.ts [hunk 1]
```diff
@@ -253,4 +256,84 @@ export class L1ToL2MessageGasEstimator {
...
+  public async populateL1ToL2FunctionParams(
+    /**
+     * Function that will internally make an L1->L2 transaction
+     * Will initially be called with dummy values to trigger a special revert containing
+     * the real params. Then called again with the real params to form the final data to be submitted
+     */
+    dataFunc: (params: {
+      gasLimit: BigNumber
+      maxFeePerGas: BigNumber
+      maxSubmissionFee: BigNumber
```
Comment (fredlacs) [inline src/lib/message/L1ToL2MessageGasEstimator.ts:277]: any reason not to use `L1ToL2MessageGasParams` as the type here?

File: src/lib/message/L1ToL2MessageGasEstimator.ts [hunk 2]
```diff
@@ -253,4 +256,84 @@ export class L1ToL2MessageGasEstimator {
...
+    })
+
+    // get retryable data from the null call
+    const res = await l1Provider.call({
+      to: to,
+      data: nullData,
+      value: value,
+      from: from,
+    })
+    const retryable = RetryableDataTools.tryParseError(res)!
```
[the remark this answers is not in the document]
Comment (fredlacs) [reply src/lib/message/L1ToL2MessageGasEstimator.ts:301]: seems like ethersjs doesn't throw an error on solidity custom errors

File: src/lib/message/L1ToL2MessageGasEstimator.ts (hunk 1 above)
Comment (yahgwai) [reply src/lib/message/L1ToL2MessageGasEstimator.ts]: No, reused now

File: src/lib/message/L1ToL2MessageGasEstimator.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (yahgwai) [reply src/lib/message/L1ToL2MessageGasEstimator.ts]: I've wrapped this in a try/catch anyway, so if in future it does throw properly we can still catch here

File: src/lib/message/L1ToL2MessageGasEstimator.ts [hunk 3]
```diff
@@ -253,4 +261,93 @@ export class L1ToL2MessageGasEstimator {
...
+        data: nullData,
+        value: value,
+        from: from,
+      })
+      retryable = RetryableDataTools.tryParseError(res)
+      if (!isDefined(retryable)) {
+        throw new ArbSdkError(`No retryable data found in error: ${res}`)
+      }
+    } catch (err) {
+      retryable = RetryableDataTools.tryParseError(err as Error)
```
Comment (fredlacs) [inline src/lib/message/L1ToL2MessageGasEstimator.ts:311]: would be good to add a comment explaining https://github.com/OffchainLabs/arbitrum-sdk/pull/149#discussion_r952677274

**Review by fredlacs — APPROVED** [review]
would be good to add a comment in for future reference, but lgtm

**Pull request merged**